### ISSUE TITLE
Some questions about the appearance space and the structure space

### ISSUE THREAD

**Issue by SY-Xuan** [NVlabs/DG-Net#37]
Hello, In your paper, I think if use appearance space of id **I** and structure space of id **J** to generate an image. The ID of generated image should be **J**. So I think the structure space encodes the ID information. 
This loss use the ID of appearance space to determine the ID of generated image.
![image](https://user-images.githubusercontent.com/22125606/77287955-bebab480-6d11-11ea-9e21-ac940067793b.png)

But this loss use the ID of structure space to determine the ID of generated image.
![image](https://user-images.githubusercontent.com/22125606/77287941-b498b600-6d11-11ea-9c1c-98eeb26fe50d.png)

And you also mentioned in the paper that 

> _When training on images organized in this manner, the discriminative module is forced to learn the fine-grained id-relate attributes (such as hair, hat, bag, body size, and so on) that are  independent to clothing._ 

And obviously the structure space encodes the hair, hat, bag and body size.  Therefore, the sutructure space encodes the ID.

This is confusing to use appearance space to discriminative ID. Could you please explain about this.

**Comment by Zonsor** [comment]
@BossBobxuan I also have same questions. My hypothesis is we need to make generated image have high predicted probability for the ID from both spaces. The reason is both have id-related information.

**Comment by layumi** [comment]
Hi @BossBobxuan , @Zonsor 
Sorry for the late response.  Yes. It is possible.

However, we did not do it. The main reason is  that the structure space is relatively low-level, which is used to reconstruct the image. Thus, if we want to extract the high-level feature, we need one more res-net, which will introduce extra parameters. 

So we conduct a tradeoff that learn the structure  info, e.g., hair, hat, bag and body size, on the appearance embedding as well.

**Comment by RonakDedhiya** [comment]
Hello,

I couldn't find in code how fine grained classification is done?
Or
how structure info is learnt in appearance embedding?

**Comment by layumi** [comment]
> Hello,
> 
> I couldn't find in code how fine grained classification is done?
> Or
> how structure info is learnt in appearance embedding?

We just use two classifiers, which do not share weights.
https://github.com/NVlabs/DG-Net/blob/master/reIDmodel.py#L145-L146

**Comment by RonakDedhiya** [comment]
So the ft_netAB has a shared base parameters for two purposes:
1. learning f - as appearance information
2. learning p - which does reID learning.

I have a naive doubt that whether base parameters will be a tradeoff between having nice appearance information and also having discriminative reID information. Can we use different model for learning p?

**Comment by layumi** [comment]
Yes. The generation somehow affect the reID. So I applied the detach at https://github.com/NVlabs/DG-Net/blob/master/reIDmodel.py#L142  
Sure. You could use different model to learning p.

**Comment by RonakDedhiya** [comment]
Thanks for your quick response. Your work is quite inspiring and I am learning a lot from your paper.

**Comment by yian2271368** [comment]
> However, we did not do it. The main reason is that the structure space is relatively low-level, which is used to reconstruct the image. Thus, if we want to extract the high-level feature, we need one more res-net, which will introduce extra parameters.

i think here you want to say " structure space is relatively high-level"?

**Comment by SY-Xuan** [comment]
> Hi @BossBobxuan , @Zonsor
> Sorry for the late response. Yes. It is possible.
> 
> However, we did not do it. The main reason is that the structure space is relatively low-level, which is used to reconstruct the image. Thus, if we want to extract the high-level feature, we need one more res-net, which will introduce extra parameters.
> 
> So we conduct a tradeoff that learn the structure info, e.g., hair, hat, bag and body size, on the appearance embedding as well.

Thanks for your response.
